compiler/renderverbatim.nim

```diff
@@ -23,6 +25,62 @@ proc isInIndentationBlock(src: string, indent: int): bool =
     if src[j] != ' ': return false
   return true
 
+type LineData = object
+  ## this avoids having to use a HashSet (but we could...)
+  lineFirst: int
+  conf: ConfigRef
+  lines: seq[bool]
+
+proc tripleStrLitStartsAtNextLine(conf: ConfigRef, n: PNode): bool =
+  # enabling TLineInfo.offsetA,offsetB would probably make this easier
+  const tripleQuote = "\"\"\""
+  let src = sourceLine(conf, n.info)
+  let col = n.info.col
+  doAssert src.continuesWith(tripleQuote, col) # sanity check
+  var i = col + 3
+  var onlySpace = true
+  while true:
+    if src.len <= i:
+      doAssert src.len == i
+      return onlySpace
+    elif src.continuesWith(tripleQuote, i) and (src.len == i+3 or src[i+3] != '\"'):
+      return false # triple lit is in 1 line
+    elif src[i] != ' ': onlySpace = false
+    i.inc
+
+proc visitMultilineStrings(data: var LineData, n: PNode) =
+  var cline = data.lineFirst
+
+  template setLine() =
+    let line2 = cline - data.lineFirst
+    if data.lines.len <= line2:
+      data.lines.setLen line2+1
+      data.lines[line2] = true
+
+  case n.kind
+  of nkTripleStrLit:
+    # same logic should be applied for any multiline token
+    # we could also consider nkCommentStmt but right now we just assume doc comments,
+    # unlike triple string litterals, don't de-indent from runnableExamples.
+    cline = n.info.line.int
+    if tripleStrLitStartsAtNextLine(data.conf, n):
+      cline.inc
+      setLine()
+    for ai in n.strVal:
+      case ai
+      of '\n':
+        cline.inc
+        setLine()
+      else: discard
+  else:
+    for i in 0..<n.safeLen:
+      visitMultilineStrings(data, n[i])
+
+proc startOfLineInsideTriple(data: LineData, line: int): bool =
+  let line2 = line - data.lineFirst
+  if line2 >= data.lines.len: false
+  else: data.lines[line2]
```

Comment:
Please try to use more descriptive variable names.

Reply:
fixed, + a few other improvements to doc comments (although in a 3 line proc, conciseness is important too)